src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagService.cs

```diff
@@ -47,24 +55,47 @@ public async Task<AddCustomTagResponse> AddCustomTagAsync(IEnumerable<CustomTagE
             // Validate input
             _customTagEntryValidator.ValidateCustomTags(customTags);
 
-            Dictionary<long, CustomTagStoreEntry> addedTags = new Dictionary<long, CustomTagStoreEntry>();
-            foreach (var tag in customTags)
+            List<CustomTagStoreEntry> addedCustomTagStoreEntries = new List<CustomTagStoreEntry>();
+            foreach (var customTag in customTags)
             {
                 try
                 {
                     CustomTagStatus initStatus = CustomTagStatus.Reindexing;
-                    long key = await _customTagStore.AddCustomTagAsync(tag.Path, tag.VR, tag.Level, initStatus);
-                    CustomTagStoreEntry storeEntry = new CustomTagStoreEntry(key, tag.Path, tag.VR, tag.Level, initStatus);
-                    addedTags.Add(key, storeEntry);
+
+                    // we only support top level dicomtag right now, will support embeded dicomtag soon.
+                    DicomTag[] tags;
+                    if (!_dicomTagParser.TryParse(customTag.Path, out tags, supportMultiple: false))
+                    {
+                        // not a valid dicom tag path
+                        throw new CustomTagEntryValidationException(
+                            string.Format(CultureInfo.InvariantCulture, DicomCoreResource.InvalidCustomTag, customTag));
+                    }
+
+                    DicomTag tag = tags[0];
+                    string path = tag.GetPath();
+
+                    string vr = customTag.VR;
+
+                    // when VR is not specified for standard tag,
+                    if (!tag.IsPrivate && tag.DictionaryEntry != DicomDictionary.UnknownTag)
+                    {
+                        if (string.IsNullOrWhiteSpace(vr))
+                        {
+                            vr = tag.GetDefaultVR()?.Code;
+                        }
+                    }
+
+                    CustomTagStoreEntry storeEntry = await _customTagStore.AddCustomTagAsync(path, vr, customTag.Level, initStatus);
+                    addedCustomTagStoreEntries.Add(storeEntry);
                 }
                 catch (Exception ex)
```

Comment:
Why are we catching and deleting here? Add should be last thing to happen.

Reply:
Actually I'm not very sure about the right solution here before know how the job framework works. That's why add several lines of TODO above.

Reply:
Since we don't have general job framework, these logic will be changed as our specific job framework comes out. Current plan is that adding custom tags and starting reindex job are done in one SQL transaction, the reindex job will reindex and update custom tag status after completion. 
The new logic should be able to solve all our concerns.

Since another PR is pending on this, and the logic here will be rewritten after our job framework is finished, it is OK to get this PR approved for now, so that dependent  code (for customTagStore) could be checked-in.